core/trino-main/src/main/java/io/trino/execution/DropViewTask.java

```diff
@@ -64,27 +65,21 @@ public String getName()
         QualifiedObjectName name = createQualifiedObjectName(session, statement, statement.getName());
 
         if (metadata.isMaterializedView(session, name)) {
-            if (!statement.isExists()) {
-                throw semanticException(
-                        TABLE_NOT_FOUND,
-                        statement,
-                        "View '%s' does not exist, but a materialized view with that name exists. Did you mean DROP MATERIALIZED VIEW %s?", name, name);
-            }
-            return immediateVoidFuture();
+            throw semanticException(
+                    GENERIC_USER_ERROR,
+                    statement,
+                    "View '%s' does not exist, but a materialized view with that name exists. Did you mean DROP MATERIALIZED VIEW %s?", name, name);
         }
 
         if (!metadata.isView(session, name)) {
-            if (!statement.isExists()) {
-                Optional<TableHandle> table = metadata.getTableHandle(session, name);
-                if (table.isPresent()) {
-                    throw semanticException(
-                            TABLE_NOT_FOUND,
-                            statement,
-                            "View '%s' does not exist, but a table with that name exists. Did you mean DROP TABLE %s?", name, name);
-                }
-                throw semanticException(TABLE_NOT_FOUND, statement, "View '%s' does not exist", name);
+            Optional<TableHandle> table = metadata.getTableHandle(session, name);
```

Comment:
In the context of connectors using shared metastores (hive, iceberg, delta) we would have this call throwing a bogus exception about saying "cannot query Iceberg/Delta/Hive table".

We actually want to know here whether the table exists and not get actually get extensive information about the table here.

I propose (in a follow-up PR) that we introduce a method `tableExists(session, name)` in `MetadataManager` as well as on `ConnectorMetadata` to  avoid such exceptions.

cc @findepi @ebyhr

Reply:
@findinpath  good catch, since `getTableHandle` returns optional,  instead of throwing if we defaulted returned an `Opional.OfEmtpy()`  is that behavior in-line with this function call?

Reply:
@Riddle4045 in the context of using a shared metastore (e.g. : say hive and iceberg use the same Hive Metastore Service) we could have the hive table `foobar` visible in both Hive & Iceberg. However if we are in the context of Iceberg (by previously using `USE iceberg.default`) and trying to drop a view in the following manner `DROP VIEW foobar` (note the unqualified name for the table) we'll get an exception saying `Cannot query Iceberg/Delta/Hive table` - which is a rather misleading  error message.

Reply:
@findinpath  Thanks for the explanation! Makes sense to me now. I'll create an issue for this :).